src/Shared/NativeMethodsShared.cs

```diff
@@ -486,14 +490,37 @@ internal static MemoryStatus GetMemoryStatus()
         internal static DateTime GetLastWriteFileUtcTime(string fullPath)
         {
             DateTime fileModifiedTime = DateTime.MinValue;
-            WIN32_FILE_ATTRIBUTE_DATA data = new WIN32_FILE_ATTRIBUTE_DATA();
-            bool success = false;
 
-            success = NativeMethodsShared.GetFileAttributesEx(fullPath, 0, ref data);
-            if (success)
+            if (UseSymlinkTimeInsteadOfTargetTime)
             {
-                long dt = ((long)(data.ftLastWriteTimeHigh) << 32) | ((long)data.ftLastWriteTimeLow);
-                fileModifiedTime = DateTime.FromFileTimeUtc(dt);
+                WIN32_FILE_ATTRIBUTE_DATA data = new WIN32_FILE_ATTRIBUTE_DATA();
+                bool success = false;
+
+                success = NativeMethodsShared.GetFileAttributesEx(fullPath, 0, ref data);
+                if (success)
+                {
+                    long dt = ((long) (data.ftLastWriteTimeHigh) << 32) | ((long) data.ftLastWriteTimeLow);
+                    fileModifiedTime = DateTime.FromFileTimeUtc(dt);
+                }
+            }
+            else
+            {
+                using (SafeFileHandle handle =
+                    CreateFile(fullPath, GENERIC_READ, FILE_SHARE_READ, IntPtr.Zero, OPEN_EXISTING,
+                        FILE_ATTRIBUTE_NORMAL, IntPtr.Zero))
+                {
+                    if (!handle.IsInvalid)
```

Comment:
Error instead of DateTime.MinValue? Or is the caller code error checking against DateTime.MinValue?